### PR TITLE
[PR] Debug adapters + filters

### LINKED ISSUE #130: [BUG] Can't add a relation when it already exists : MISSING ERROR
**Description of the bug**
Can't add a relation when it already exists : MISSING ERROR 
=> add a simple warning like : the relation already exists

### LINKED ISSUE #132: [BUG] Crash when no relation Type (+warning relation type is required)
**Description of the bug**
can't edit relation details when first pushing it null

### LINKED ISSUE #135: [BUG] Values don't remain when editing
**Description of the bug**
When editing a person, the values don't remain, you have to enter everything again

### PR DESCRIPTION
**Pull request by mathisfchx**
### Description
New adapters to correctly update when new data
Filters features now working with the new LinkedList
Some debug on relation

### CODE REVIEW

File: Netw4ppl/app/src/main/java/netw4ppl/ines/utils/PersonListAdapter.java [hunk 1]
```diff
@@ -192,40 +225,44 @@ public View getView(int position, @Nullable View convertView, @NonNull ViewGroup
...
-                                holder.mDescriptiveField1.setText(String.valueOf(person.getInfoByKey(key)));
-                            }
-                        } else {
-                            holder.mDescriptiveField2Title.setText(field + " : ");
-                            if (!field.getLinkedList().equals("")) {
-                                linkedListValue = field.getLinkedListValue(person);
-                            }
-                            if (linkedListValue != null) {
-                                holder.mDescriptiveField2.setText(linkedListValue);
+            if (usedFields.containsKey(person)) {
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/utils/PersonListAdapter.java:228]: beurk 🤢

File: Netw4ppl/app/src/main/java/netw4ppl/ines/AddRelationActivity.java [hunk 2]
```diff
@@ -342,6 +340,7 @@ private boolean samePersonRelation(Relation relation){
     private boolean testExistingRelation(Relation relation){
         for(int i=0 ; i<ManageRelationsActivity.array_relations.size();i++){
             if (relation.isSameRelation(ManageRelationsActivity.array_relations.get(i))){
+                System.out.println("Relation already exists");
```
Comment (lduf) [inline Netw4ppl/app/src/main/java/netw4ppl/ines/AddRelationActivity.java]:
```suggestion
                
```

**Review by lduf — CHANGES_REQUESTED** [review]

**Review by lduf — APPROVED** [review]
🌍

File: Netw4ppl/app/src/main/java/netw4ppl/ines/utils/PersonListAdapter.java (hunk 1 above)
Comment (mathisfchx) [reply Netw4ppl/app/src/main/java/netw4ppl/ines/utils/PersonListAdapter.java:228]: :'( #128

**Pull request merged**